File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/carpenterswall/domain/CarpentersWallUpObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.carpenterswall.domain;

import com.zwstudio.logicpuzzlesandroid.home.domain.HintState;

public class CarpentersWallUpObject extends CarpentersWallObject {
    public HintState state = HintState.Normal;
    public boolean isHint() {return true;}
    public String objAsString() {
        return "up";
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/carpenterswall/domain/CarpentersWallUpObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.carpenterswall.domain

import com.zwstudio.logicpuzzlesandroid.home.domain.HintState

class CarpentersWallUpObject : CarpentersWallObject() {
    var state = HintState.Normal
    override val isHint: Boolean
        get() = true

    override fun objAsString(): String {
        return "up"
    }
}